R/expect_lint.R

```diff
@@ -88,10 +89,10 @@ expect_lint <- function(content, checks, ..., file = NULL, language = "en") {
         itr <<- itr + 1L
         lapply(names(check), function(field) {
           if (!field %in% lint_fields) {
-            stop(sprintf(
-              "check #%d had an invalid field: \"%s\"\nValid fields are: %s\n",
-              itr, field, toString(lint_fields)
-            ), call. = FALSE)
+            cli_abort(c(
+              x = "Check {.val {itr}} has an invalid field: {.field {field}}.",
+              i = "Valid fields are: {.field {toString(lint_fields)}}."
```

Comment:
```suggestion
              i = "Valid fields are: {.field {lint_fields}}."
```

{cli} auto-aggregates, see https://cli.r-lib.org/articles/pluralization.html#use-the-length-of-character-vectors

After this change, I get:

```
Error:
✖ Check 1 has an invalid field: abc.
ℹ Valid fields are: filename, line_number, column_number, type, message, line, ranges, and linter.
Run `rlang::last_trace()` to see where the error occurred.
```

The coloring is also nicer, instead of coloring the commas that `toString()` produces, on the actual `lint_field` text is colored.

Reply:
Note that by default {cli} adds `...` to truncate vectors >20 elements, if we want to avoid that it's pretty clunky:

https://cli.r-lib.org/reference/inline-markup.html#collapsing-inline-vectors

Reply:
Will that be an issue? Do we expect any relevant vector in this context to contain more than 20 elements? 

The only one that I have encountered that has _many_ elements is related to supported R versions, which has length of 11. Otherwise, nothing comes to mind.

Reply:
I agree it's not a major concern, just something to be aware of.